test/socket_test.dart

```diff
@@ -112,4 +113,52 @@ void main() {
 
     expect(factoryCalls, 1);
   });
+
+  test(
+    'socket connect keeps retrying if an exception is thrown while building the mount point',
+    () {
+      final mockPhoenixSocketOptions = MockPhoenixSocketOptions();
+      when(mockPhoenixSocketOptions.reconnectDelays).thenReturn(const [
+        Duration.zero,
+        Duration(seconds: 10),
+        Duration(seconds: 20),
+        Duration(seconds: 30),
+      ]);
+      when(mockPhoenixSocketOptions.getParams()).thenThrow(Exception());
+
+      final phoenixSocket = PhoenixSocket(
+        'ws://endpoint',
+        socketOptions: mockPhoenixSocketOptions,
+      );
+
+      fakeAsync(
+        (async) {
+          phoenixSocket.connect();
+
+          verify(mockPhoenixSocketOptions.getParams()).called(1);
+          expect(phoenixSocket.isConnected, isFalse);
+
+          // first retry after ~10 seconds
+          async.elapse(const Duration(seconds: 11));
```

Comment:
Time doesn't flow by itself in `fakeAsync`, you should be able to use exact durations from reconnectDelays. (also below)

Reply:
The delay used is coming from the list of reconnect delays passed in options plus a random duration between 0 and 1 second. That's why I went with custom reconnect delays where the 0-1 random second does not get in the way.